docs/source/cpp/compute.rst

```diff
@@ -450,6 +450,18 @@ Mixed time resolution temporal inputs will be cast to finest input resolution.
 +------------------+--------+------------------+----------------------+-------+
 | divide_checked   | Binary | Numeric/Temporal | Numeric/Temporal     | \(1)  |
 +------------------+--------+------------------+----------------------+-------+
+| exp              | Unary  | Numeric          | Numeric              |       |
```

Comment:
We should be more precise when listing input and output types (see examples under "Logaritmic Functions" below).
Also you'll need to correct the table formatting after this :-)
```suggestion
| exp              | Unary  | Float32/Float64/Decimal | Float32/Float64      |       |
```

Reply:
Wouldn't integer be a valid input type and hence input type remains `Numeric`? `Float32/Float64` as output makes sense.

This is out of scope for this PR, but how about `complex` values ([example](https://numpy.org/devdocs/reference/generated/numpy.exp.html))? Should Acero kernels support complex? What about Substrait?

Reply:
I went with your suggestion, but I think we should probably still specify integers?

Reply:
I don't know, does `exp` actually accept integers?

Reply:
As per [cppreference](https://en.cppreference.com/w/cpp/numeric/math/exp) it does since c++11.

Reply:
I'll admit it's not a terribly interesting case but still.

Reply:
I'm not asking whether C++ supports it, but whether Arrow wants to support it :-)

As a comparison, our logarithm functions only support reals (either floating-point or decimals), not integers.

Reply:
Integer support for logarithms feels more important than here. But I suppose implicit casting will cover the gap if needed? Shall I remove the integer support then?

Reply:
Oh we're using `MakeUnaryArithmeticFunctionFloatingPoint` we can't remove it anyway.